controllers/perconaservermysql_controller.go

```diff
@@ -1531,3 +1539,179 @@ func (r *PerconaServerMySQLReconciler) restartGroupReplication(ctx context.Conte
+			return err
+		}
+	}
+	hosts := []string{
+		fmt.Sprintf("*.%s-mysql", cr.Name),
+		fmt.Sprintf("*.%s-mysql.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-mysql.%s.svc.cluster.local", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator", cr.Name),
+		fmt.Sprintf("*.%s-orchestrator.%s", cr.Name, cr.Namespace),
+		fmt.Sprintf("*.%s-orchestrator.%s.svc.cluster.local", cr.Name, cr.Namespace),
+	}
+	kubeCert := &cm.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            cr.Name + "-ssl",
+			Namespace:       cr.Namespace,
+			OwnerReferences: ownerReferences,
+		},
+		Spec: cm.CertificateSpec{
+			SecretName: cr.Spec.SSLSecretName,
+			DNSNames:   hosts,
+			IsCA:       false,
+			IssuerRef: cmmeta.ObjectReference{
+				Name:  issuerName,
+				Kind:  issuerKind,
+				Group: issuerGroup,
+			},
+		},
+	}
+
+	if cr.Spec.TLS != nil && len(cr.Spec.MySQL.ServiceAccountName) > 0 {
+		kubeCert.Spec.DNSNames = append(kubeCert.Spec.DNSNames, cr.Spec.TLS.SANs...)
+	}
+
+	err = r.Create(context.TODO(), kubeCert)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "create certificate")
+	}
+
+	return r.waitForCerts(cr.Namespace, cr.Spec.SSLSecretName)
+}
+
+func (r *PerconaServerMySQLReconciler) waitForCerts(namespace string, secretsList ...string) error {
+	ticker := time.NewTicker(3 * time.Second)
+	timeoutTimer := time.NewTimer(30 * time.Second)
+	defer timeoutTimer.Stop()
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeoutTimer.C:
+			return errors.Errorf("timeout: can't get tls certificates from certmanager, %s", secretsList)
+		case <-ticker.C:
+			sucessCount := 0
+			for _, secretName := range secretsList {
+				secret := &corev1.Secret{}
+				err := r.Get(context.TODO(), types.NamespacedName{
+					Name:      secretName,
+					Namespace: namespace,
+				}, secret)
+				if err != nil && !k8serrors.IsNotFound(err) {
+					return err
+				} else if err == nil {
+					sucessCount++
+				}
+			}
+			if sucessCount == len(secretsList) {
+				return nil
+			}
+		}
+	}
+}
+
+func (r *PerconaServerMySQLReconciler) createIssuer(ownRef []metav1.OwnerReference, namespace, issuer string, caCertSecret string) error {
+
+	spec := cm.IssuerSpec{}
+
+	if caCertSecret == "" {
+		spec = cm.IssuerSpec{
+			IssuerConfig: cm.IssuerConfig{
+				SelfSigned: &cm.SelfSignedIssuer{},
+			},
+		}
+	} else {
+		spec = cm.IssuerSpec{
+			IssuerConfig: cm.IssuerConfig{
+				CA: &cm.CAIssuer{SecretName: caCertSecret},
+			},
+		}
+	}
+
+	err := r.Create(context.TODO(), &cm.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            issuer,
+			Namespace:       namespace,
+			OwnerReferences: ownRef,
+		},
+		Spec: spec,
+	})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "create issuer")
+	}
+	return nil
+}
+
+// OwnerRef returns OwnerReference to object
+func OwnerRef(ro runtime.Object, scheme *runtime.Scheme) (metav1.OwnerReference, error) {
```

Comment:
I don't think we need this function, there is a helper function from controller-runtime: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/125/commits/823d6fd2be6512f64962dcd862c4ac3aac7766e8